src/ipahealthcheck/ipa/trust.py

```diff
@@ -0,0 +1,499 @@
+        if not self.registry.trust_controller:
+            logger.debug('Not a trust controller, skipping')
+            return
+
+        agent_dn = DN(('krbprincipalname',
+                      'cifs/%s@%s' % (api.env.host, api.env.realm)),
+                      api.env.container_service, api.env.basedn)
+        group_dn = DN(('cn', 'adtrust agents'), api.env.container_sysaccounts,
+                      api.env.basedn)
+        try:
+            entry = self.conn.get_entry(
+                agent_dn,
+                attrs_list=['memberOf'])
+        except Exception as e:
+            yield Result(self, constants.ERROR,
+                         key=str(agent_dn),
+                         error=str(e),
+                         msg='Error retrieving ldap entry {key}: '
+                         '{error}')
+        else:
+            memberof = entry.get('memberof', [])
+            for member in memberof:
+                if DN(member) == group_dn:
+                    yield Result(self, constants.SUCCESS,
+                                 key='cifs/%s@%s' %
+                                 (api.env.host, api.env.realm))
+                    return
+
+            yield Result(self, constants.ERROR,
+                         key='cifs/%s@%s' % (api.env.host, api.env.realm),
+                         group='adtrust agents',
+                         msg='{key} is not a member of {group}')
+
+
+@registry
+class IPATrustControllerServiceCheck(IPAPlugin):
+    """
+    Verify that the current host starts the ADTRUST service.
+    """
+    @duration
+    def check(self):
+        if not self.registry.trust_controller:
+            logger.debug('Not a trust controller, skipping')
+            return
+
+        service_dn = DN(('cn', 'ADTRUST'), ('cn', api.env.host),
+                        api.env.container_masters, api.env.basedn)
+
+        try:
+            entry = self.conn.get_entry(
+                service_dn,
+                attrs_list=['ipaconfigstring'])
+        except Exception as e:
+            yield Result(self, constants.ERROR,
+                         key=str(service_dn),
+                         error=str(e),
+                         msg='Error retrieving ldap entry {key}: '
+                         '{error}')
+        else:
+            configs = entry.get('ipaconfigstring', [])
+            enabled = False
+            for config in configs:
+                if config == ENABLED_SERVICE:
+                    enabled = True
+                    break
+
+            if enabled:
+                yield Result(self, constants.SUCCESS,
+                             key='ADTRUST')
+            else:
+                yield Result(self, constants.ERROR,
+                             key='ADTRUST',
+                             msg='{key} service is not enabled')
+
+
+@registry
+class IPATrustControllerConfCheck(IPAPlugin):
+    """
+    Verify that smb.conf matches the template
+    """
+    @duration
+    def check(self):
```

Comment:
It is just fine to have more options in the `net conf list` than what we have in the `smb.conf.template`.  Also, some options aren't really important to see exact. For example, `log level` might be any. I'm not sure how to achieve this...

Reply:
Should we punt on this then? Perhaps we can come up with a list of absolutely-required settings that I can validate.

Reply:
well, there are few key ones. We can do them in a separate pull request. I'll build a list.